airsonic-main/src/main/java/org/airsonic/player/service/search/AnalyzerFactory.java

```
@@ -1,121 +1,182 @@
+                    "pattern", "\\)", "replacement", " ", "replace", "all")
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\{\\}", "replacement", "\\{ \\}", "replace", "all")
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\[\\]", "replacement", "\\[ \\]", "replace", "all");
+    }
+
+    private Builder createDefaultAnalyzerBuilder() throws IOException {
+        Builder builder = CustomAnalyzer.builder()
+                .withTokenizer(StandardTokenizerFactory.class)
+                .addTokenFilter(CJKWidthFilterFactory.class)
+                .addTokenFilter(ASCIIFoldingFilterFactory.class, "preserveOriginal", "false")
+                .addTokenFilter(LowerCaseFilterFactory.class)
+                .addTokenFilter(StopFilterFactory.class)
+                .addTokenFilter(EnglishPossessiveFilterFactory.class);
+        addTokenFilterForUnderscoreRemovalAroundToken(builder);
+        return builder;
+    }
+
+    private Builder createKeywordAnalyzerBuilder() throws IOException {
+        return CustomAnalyzer.builder()
+                .withTokenizer(KeywordTokenizerFactory.class);
+    }
+
+    private Builder createGenreAnalyzerBuilder() throws IOException {
+        Builder builder = createKeywordAnalyzerBuilder();
+        addTokenFilterForTokenToDomainValue(builder);
+        return builder;
+    }
+
+    /**
+     * Returns the Analyzer to use when generating the index.
+     * 
+     * Whether this analyzer is applied to input values ​​depends on
+     * the definition of the document's fields.
+     * 
+     * @return analyzer for index
+     * @see DocumentFactory
+     */
+    public Analyzer getAnalyzer() {
+        if (isEmpty(analyzer)) {
+            try {
+                analyzer = createDefaultAnalyzerBuilder().build();
+            } catch (IOException e) {
+                LOG.error("Error when initializing Analyzer.", e);
```

Why discard the error here? Maybe wrap it with a runtime exception and let it bubble up?

---

I revised that according to your advice.